### DISCUSSION TITLE
Set logger when resolving logger aware classes

### DISCUSSION THREAD

**Discussion by jmsfwk** [laravel/framework#49910] [Ideas]
Add support for setting a logger when resolving classes that implement `LoggerAwareInterface`.

I know that I could require a logger in a constructor, or in any other method where Laravel resolves dependencies, but sometimes I don't _need_ a logger it's just a nice to have.

I can handle not having one much more easily now that I can use the null safe operator, and therefore use logger calls like comments.

```php
public function example()
{
    if (/* some condition */) {
        $this->logger?->info('Some condition is x, so y');
        return;
    }
}
```

**Comment by rojtjo** [comment]
You could register a resolving callback to do this, but since you mention that you sometimes don't need a logger, you'd have to add some additional logic to determine whether to inject the logger or not.

```php
use Illuminate\Foundation\Application;
use Illuminate\Support\ServiceProvider;
use Psr\Log\LoggerAwareInterface;
use Psr\Log\LoggerInterface;

class AppServiceProvider extends ServiceProvider
{
    public function register(): void
    {
        $this->app->resolving(LoggerAwareInterface::class, function (LoggerAwareInterface $object, Application $app) {
            $object->setLogger($app->make(LoggerInterface::class));
        });
    }
}
```
